packages/store/ts/config/v2/schema.ts

```diff
@@ -0,0 +1,23 @@
+import { evaluate } from "@arktype/util";
+import { AbiTypeScope } from "./scope";
+
+export type SchemaInput<scope extends AbiTypeScope = AbiTypeScope> = {
+  [key: string]: keyof scope["validTypes"];
+};
+
+export type resolveSchema<schema extends SchemaInput<scope>, scope extends AbiTypeScope> = evaluate<{
+  [key in keyof schema]: {
+    /** the Solidity primitive ABI type */
+    type: scope["validTypes"][schema[key]];
+    /** the user defined type or Solidity primitive ABI type */
+    internalType: schema[key];
+  };
+}>;
+
+export function resolveSchema<schema extends SchemaInput<scope>, scope extends AbiTypeScope = AbiTypeScope>(
+  schema: schema,
+  scope?: scope
```

Comment:
wondering about naming this arg `abiTypeScope`

Reply:
`abiTypeScope` is just the default value but the scope can include more than just abi types

Reply:
`scope` is basically "all types that are in scope here"